maze_transformer/generation/lattice_maze.py

```diff
@@ -143,70 +143,56 @@ def find_shortest_path(
         self,
         c_start: CoordTup,
         c_end: CoordTup,
-    ) -> list[Coord]:
+    ) -> list[CoordTup]:
         """find the shortest path between two coordinates, using A*"""
 
-        g_score: dict[
-            CoordTup, float
-        ] = dict()  # cost of cheapest path to node from start currently known
-        f_score: dict[CoordTup, float] = {
-            c_start: 0.0
-        }  # estimated total cost of path thru a node: f_score[c] := g_score[c] + heuristic(c, c_end)
-
-        # init
-        g_score[c_start] = 0.0
-        g_score[c_start] = self.heuristic(c_start, c_end)
-
-        closed_vtx: set[CoordTup] = set()  # nodes already evaluated
-        open_vtx: set[CoordTup] = set([c_start])  # nodes to be evaluated
-        source: dict[
-            CoordTup, CoordTup
-        ] = (
-            dict()
-        )  # node immediately preceding each node in the path (currently known shortest path)
-
-        while open_vtx:
+        # cheapest currently known path to node from start
+        g_score: dict[CoordTup, float] = {c_start: 0.0}
+        # estimated total cost of path through a node: f_score[c] = g_score[c] + heuristic(c, c_end)
+        f_score: dict[CoordTup, float] = {c_start: self.heuristic(c_start, c_end)}
```

Comment:
Here is the part in the new version where I have changed behaviour (see PR description for details)